clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -303,6 +302,19 @@ private void throwIfOffsetOutOfRange() throws OffsetOutOfRangeException {
     }
 
     /**
+     * If any topic from previous fetchResponse contatains Authorization error, throw ApiException.
+     * @throws ApiException
+     */
+    private void throwIfUnauthorized() throws ApiException {
```

Comment:
The correct behavior is that if you don't have Describe or Read access you should get back an AuthorizationException. Currently I handled the Read part but the consumer metadata waiting forever issue is still present and I believe we have a followup jira to fix it. We need to ensure that the metadata receiver throws the exception back to the caller when a nonRetriableException is thrown.